src/main/java/br/com/dandrade/viagens/controllers/dto/output/StretchDto.java

```diff
@@ -0,0 +1,62 @@
+package br.com.dandrade.viagens.controllers.dto.output;
+
+import br.com.dandrade.viagens.models.AirRoute;
+import br.com.dandrade.viagens.models.Stretch;
+import br.com.dandrade.viagens.models.StretchMinimalInfo;
+
+import java.time.LocalTime;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.List;
+
+public class StretchDto {
+    private String out;
+    private String in;
+    private LocalTime takeOff;
+    private LocalTime landing;
+
+    public StretchDto( AirRoute airRoute,  LocalTime takeOff, LocalTime landing ) {
+        this(airRoute.getOriginAirportName(), airRoute.getDestinyAirportName(), takeOff, landing);
+    }
+
+    public StretchDto( String out, String in, LocalTime takeOff, LocalTime landing ) {
+        this.out = out;
+        this.in = in;
+        this.takeOff = takeOff;
+        this.landing = landing;
+    }
+
+
+    public String getOut() {
+        return out;
+    }
+
+    public String getIn() {
+        return in;
+    }
+
+    public LocalTime getTakeOff() {
+        return takeOff;
+    }
+
+    public LocalTime getLanding() {
+        return landing;
+    }
+
+    public static List<StretchDto> create( StretchMinimalInfo minimalInfo ) {
+
+        List<StretchDto> stretchDtos = new ArrayList<>(  );
+        LocalTime takeOff = minimalInfo.getDeparture().toLocalTime();
+
+        for (Stretch stretch: minimalInfo.getStretchs()) {
```

Comment:
Aqui já reduziu bastante o que precisa para entender o código, mas eu bati o olho e pensei, ainda falta alguma coisa: existe uma lógica sobre o conjunto de paradas, tanto que você altera o valor da variável dentro do for, o que, em geral tende a complicar um pouco. Dado que as variáveis de controle meio que ficam declaradas nele mesmo. 

Minha sugestão: existir uma abstração sobre o conjunto de paradas. Imagina algo assim:

```
Collection stretchDtos  = paradas.map((route,horarioSaida, horarioChegada) -> StretchDto :: new)
```
A inteligência ia para o model e saia do DTO.